### PR TITLE
Add unique constraint to public_value

### PR DESCRIPTION
**Pull request by hibooboo2**

### CODE REVIEW

File: resources/content/db/core-061.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+    <property name="mediumtext" value="TEXT" dbms="postgresql" />
+    <property name="mediumtext" value="MEDIUMTEXT" />
+    <changeSet author="wizardofmath (generated)" id="dump1">
+        <modifyDataType columnName="public_value" newDataType="varchar(100)" tableName="credential"/>
```
Comment (ibuildthecloud) [inline resources/content/db/core-061.xml]: Change to 255

File: resources/content/db/core-061.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    </changeSet>
+    <changeSet author="wizardofmath (generated)" id="dump2">
+        <addUniqueConstraint columnNames="public_value" constraintName="public_value_UNIQUE" deferrable="false" disabled="false" initiallyDeferred="false" tableName="credential"/>
```
Comment (ibuildthecloud) [inline resources/content/db/core-061.xml]: constraintName should be idx_credential_public_value

**Comment by alena1108** [comment]
Is it possible that users' existing setups have duplicated values for the field we are applying the constraint  to?

**Comment by vincent99** [comment]
@alena1108 we talked about that [here](https://github.com/rancher/cattle/pull/810#discussion_r38247397), though I never got an answer.  If it's recoverable I think it can just be put in as a footnote in the release notes because the only people with anything other than 20-char random strings in the publicValue are probably us.

**Comment by alena1108** [comment]
@vincent99 yes, I remember that, just wanted to bring up the question up again. We should for sure release note it.

**Comment by ibuildthecloud** [comment]
As discussed this currently isn't possible due to registry credentials.

**Pull request closed without merging**